### PR TITLE
9/12 blogpost

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
Last Week in ManageIQ: Bootstrap Tree is Here

### CODE REVIEW

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+title: "Last Week in ManageIQ: Bootstrap Tree is Here"
+date: 2016-09-12
+author: zita
+comments: true
+published: true
+---
+
+Brno office sends its regards once again. I, [Zita](https://github.com/ZitaNemeckova) will tell you more about this week's news in ManageIQ codebase :)
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: Possibly s/this/last/ and "in the ManageIQ codebase"?

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Anyone can be a ManageIQ contributor, it's really easy :) . You just pick an issue from the [issue tracker](https://github.com/ManageIQ/manageiq/issues) and make a PR. 
+
+
+## Improved
+If you ever wondered why trees are looking way better this week then they did before, it's thanks to [David](https://github.com/skateman) and his [Replaced dynatree with a modified version of bootstrap-treeview](https://github.com/ManageIQ/manageiq/pull/10767) PR that he worked on lately.
+As they can grow to sequoia-like height sometimes, their performance must be top-notch. [Keenan](https://github.com/kbrock) is making sure they are as fast as they can be with [TreeBuilder use settings not config](https://github.com/ManageIQ/manageiq/pull/11138)
+and [Tree builder misc performance fixes](https://github.com/ManageIQ/manageiq/pull/11139). And more PRs are coming soon. 
+
+In the past, you had to set ports one by one but [JoeS](https://github.com/selmanj) made an improvement [Use a range of port numbers instead of a single port for load balancers](https://github.com/ManageIQ/manageiq/pull/10804) that allows you to use a range to set all the ports at once from now on. Isn't it neat?
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/JoeS/Joe S/

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Fixed
+Last week, you saw how i18n is done correctly, this week [JoeR](https://github.com/jrafanie) shows in his numerous PRs [here](https://github.com/ManageIQ/manageiq/pull/11079), [here](https://github.com/ManageIQ/manageiq/pull/11077), [here](https://github.com/ManageIQ/manageiq/pull/11074) and [here](https://github.com/ManageIQ/manageiq/pull/11071)
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/JoeR/Joe R/ (or even full name?)

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+and [Tree builder misc performance fixes](https://github.com/ManageIQ/manageiq/pull/11139). And more PRs are coming soon. 
+
+In the past, you had to set ports one by one but [JoeS](https://github.com/selmanj) made an improvement [Use a range of port numbers instead of a single port for load balancers](https://github.com/ManageIQ/manageiq/pull/10804) that allows you to use a range to set all the ports at once from now on. Isn't it neat?
+
+## Fixed
+Last week, you saw how i18n is done correctly, this week [JoeR](https://github.com/jrafanie) shows in his numerous PRs [here](https://github.com/ManageIQ/manageiq/pull/11079), [here](https://github.com/ManageIQ/manageiq/pull/11077), [here](https://github.com/ManageIQ/manageiq/pull/11074) and [here](https://github.com/ManageIQ/manageiq/pull/11071) 
+how to use interpolation correctly so that Rubocop is happy and can give you a cookie.
+
+[Simon](https://github.com/isimluk) fixed a problem with trailing nonsense in an URL with the PR [Api should return http:404 when URL has trailing garbage](https://github.com/ManageIQ/manageiq/pull/11058). So now you just get a nice 404 if you have some unwanted leftovers in your URL.
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/Api/API/

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+## New
+From now on, ManageIQ has a new notification drawer done by [Simon](https://github.com/isimluk) and [Jeffrey](https://github.com/jeff-phillips-18). The first one provided backend [Api for Notification Drawer](https://github.com/ManageIQ/manageiq/pull/10696) after hours of head scratching and the second made all that visible in the UI [Add Notifications Drawer and Toast Notifications List](https://github.com/ManageIQ/manageiq/pull/10955) (check it to see how it looks).
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/Api/API/

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+## New
+From now on, ManageIQ has a new notification drawer done by [Simon](https://github.com/isimluk) and [Jeffrey](https://github.com/jeff-phillips-18). The first one provided backend [Api for Notification Drawer](https://github.com/ManageIQ/manageiq/pull/10696) after hours of head scratching and the second made all that visible in the UI [Add Notifications Drawer and Toast Notifications List](https://github.com/ManageIQ/manageiq/pull/10955) (check it to see how it looks). 
+
+Topology graph has a new feature from [Jiri](https://github.com/Jiri-Kremser). Now you can see middleware messaging entities as well thanks to the PR [Displaying middleware messaging entities in the topology graph](https://github.com/ManageIQ/manageiq/pull/10892).
+
+Another provider related new feature is the OpenShift Deployment Wizard (bunch of new windows) added by [Daniel](https://github.com/dkorn) in the PR [OpenShift Deployment Wizard](https://github.com/ManageIQ/manageiq/pull/7620). Deployment is now easier and more user friendly.
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/provider related/provider-related/

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+Cockpit had been on [MartinP](https://github.com/martinpovolny)'s mind lately. He brought new API to connect ManageIQ with it, in the PR [Add cockpit support to the API](https://github.com/ManageIQ/manageiq/pull/10871).
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: s/new/a new/ possibly?

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+[Alberto](https://github.com/abellotti) created a new class MiqAeBrowser in the PR [Adding the MiqAeBrowser](https://github.com/ManageIQ/manageiq/pull/9586) that adds an enhancement to Automate. It brings arbitrary depth tree search of the Automate mode, support serialized objects to include `fqname`, `domain_fqname` and `klass` and more.  
+ 
+## Deleted
+[JoeR](https://github.com/jrafanie) shared his love for deleting (in this case commenting out) bad code with the PR [Fix unreachable code by removing it or commenting it out](https://github.com/ManageIQ/manageiq/pull/11110). It's always nice to delete unreachable code unless 
+you make a mistake and it crashes (personal experience).
+ 
+[Tim](https://github.com/imtayadeway) didn't delete much in his PR [Remove "typed_target" from API update paths](https://github.com/ManageIQ/manageiq/pull/11034) but nicely renamed some methods and used `super`. The functionality is the same but code is way better looking. A great example how to code better.
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: :heart: :heart: :heart:

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+Topology graph has a new feature from [Jiri](https://github.com/Jiri-Kremser). Now you can see middleware messaging entities as well thanks to the PR [Displaying middleware messaging entities in the topology graph](https://github.com/ManageIQ/manageiq/pull/10892).
+
+Another provider related new feature is the OpenShift Deployment Wizard (bunch of new windows) added by [Daniel](https://github.com/dkorn) in the PR [OpenShift Deployment Wizard](https://github.com/ManageIQ/manageiq/pull/7620). Deployment is now easier and more user friendly.
+
+Cockpit had been on [MartinP](https://github.com/martinpovolny)'s mind lately. He brought new API to connect ManageIQ with it, in the PR [Add cockpit support to the API](https://github.com/ManageIQ/manageiq/pull/10871).
+
+[Alberto](https://github.com/abellotti) created a new class MiqAeBrowser in the PR [Adding the MiqAeBrowser](https://github.com/ManageIQ/manageiq/pull/9586) that adds an enhancement to Automate. It brings arbitrary depth tree search of the Automate mode, support serialized objects to include `fqname`, `domain_fqname` and `klass` and more.
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: Not sure, but is this supposed to be Automate model?
s/support/supports/

**Comment by JPrause** [comment]
@himdel @chrisarcand can you review for tomorrow's merge.
Thanks @imtayadeway for also reviewing.

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+ 
+[Tim](https://github.com/imtayadeway) didn't delete much in his PR [Remove "typed_target" from API update paths](https://github.com/ManageIQ/manageiq/pull/11034) but nicely renamed some methods and used `super`. The functionality is the same but code is way better looking. A great example how to code better.
+
+As we moved from dynatree to bootstrap treeview, [Happy Martin](https://github.com/himdel) got rid of some old dynatree pictures in the PR [Remove remaining dynatree images](https://github.com/ManageIQ/manageiq/pull/11042) and [David](https://github.com/skateman) hid his removal of dynatree related code in his "too big for one commit" PR [Replaced dynatree with a modified version of bootstrap-treeview](https://github.com/ManageIQ/manageiq/pull/10767). 
+
+## Wrapping up
+
+I'm sure that I didn't talk about all the PRs that deserved to be mentioned this week. You can check them all out [here][PRs merged last week]. It was nice to be your guide this week :) See you next time. 
+
+[PRs merged last week]: https://github.com/ManageIQ/manageiq/pulls?utf8=✓&q=is%3Apr%20is%3Amerged%20base%3Amaster%20merged%3A%222016-09-05%20..%202016-09-12%22%20sort%3Acreated-desc%20
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: Should this, and the following link, go up to 2016-09-11?

Comment (himdel) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: Agreed, the consensus was that we go monday to sunday, so only until 09-11

**Comment by imtayadeway** [comment]
@ZitaNemeckova looks great to me! Just a few suggestions above :cake:

**Comment by ZitaNemeckova** [comment]
@imtayadeway Thanks a lot :) It should be all corrected now :)

**Comment by himdel** [comment]
Yea from me :+1: :fish_cake:

File: source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+author: zita
+comments: true
+published: true
+---
+
+Brno office sends its regards once again. I, [Zita](https://github.com/ZitaNemeckova) will tell you more about last week's news in ManageIQ codebase :)
+
+## Stats
+
+Last week, 45 contributors got [115 PRs][PRs merged last week] with [368 commits][Commits merged last week] merged to the master branch. And I will tell you more about the most interesting ones (from my point of view).
```
Comment (imtayadeway) [inline source/blog/2016-09-12-last-week-in-manageiq-bootstrap-tree.html.md]: I make it 359 commits - can you check this?

**Comment by ZitaNemeckova** [comment]
All comments should be resolved :) Thanks again guys :)

**Comment by JPrause** [comment]
@Fryguy @chessbyte here is the latest Last Week in ManageIQ blog entry.
It has been reviewed and updated.
Please merge.

Thanks @ZitaNemeckova for your submission.

**Pull request merged**